static/js/publisher/release/components/releasesConfirmDetails/index.js

```diff
@@ -27,14 +43,83 @@ const ReleasesConfirmDetails = ({ updates, isProgressiveReleaseEnabled }) => {
   const showNewReleases = Object.keys(newReleases).length > 0;
   const showPendingCloses = pendingCloses.length > 0;
 
+  const toggleGlobal = () => {
+    const newUseGlobal = !useGlobal;
+    setGlobal(newUseGlobal);
+    if (!newUseGlobal) {
+      setTemporaryProgressiveReleaseKeys();
+    } else {
+      removeTemporaryProgressiveReleaseKeys();
+    }
+  };
+
+  const updatePercentage = percentage => {
+    setGlobalPercentage(percentage);
+    updateProgressiveReleasePercentage(null, percentage);
+  };
+
   return (
     <div className="p-releases-confirm__details">
+      {showProgressiveReleases && (
+        <div className="p-releases-confirm__details-global row u-vertically-center">
```

Comment:
Could this whole thing be a separate component just rendered here? It adds quite a lot to the JSX size.